packages/dd-trace/src/opentelemetry/tracer.js

```diff
@@ -118,6 +119,10 @@ class Tracer {
   getActiveSpanProcessor () {
     return this._tracerProvider.getActiveSpanProcessor()
   }
+
+  getSpanLimits() {
```

Comment:
can we have a comment to explain why we need this method that isn't used anywhere in the code ?

Reply:
@simon-id Hello, and thank you for taking the time to review my code.

I have explained this issue in detail in the comment on issue #1244: [GitHub Issue #1244 Comment](https://github.com/DataDog/dd-trace-js/issues/1244#issuecomment-2200401180).

In summary, the Span class in @opentelemetry/sdk-trace-base calls the getSpanLimits() function of the parentTracer in its constructor. However, when the ddtrace's TraceProvider (which implements the tracer interface from opentelemetry/api) is provided as the parentTracer, it results in an error because getSpanLimits() is not defined in the TraceProvider.


Here's an example to illustrate this:

```javascript
import { registerInstrumentations } from '@opentelemetry/instrumentation';
import type { SpanLimits } from '@opentelemetry/sdk-trace-base';
import { PrismaInstrumentation } from '@prisma/instrumentation';
import tracer from 'dd-trace';

import Tracer from 'dd-trace/packages/dd-trace/src/opentelemetry/tracer';

const _spanLimits: SpanLimits = {};
Tracer.prototype.getSpanLimits = function getSpanLimits(): SpanLimits {
  return _spanLimits;
};

export const ddTracer = tracer.init();
const provider = new ddTracer.TracerProvider();

registerInstrumentations({
  tracerProvider: provider,
  instrumentations: [new PrismaInstrumentation()],
});

provider.register();
``` 

In this example, the constructor of PrismaInstrumentation calls the Span constructor from @opentelemetry/sdk-trace-base instead of @opentelemetry/api. To support Prisma instrumentation in ddtrace, the getSpanLimits() function needs to be implemented.

The reason getSpanLimits() is necessary is because @opentelemetry/sdk-trace-base relies on it to configure span limits such as the maximum number of attributes, events, and links that a span can have. Without this function, any tracer provided to OpenTelemetry instrumentations that call getSpanLimits() will fail, leading to errors during span creation. By adding getSpanLimits() to the Tracer prototype, we ensure compatibility and prevent these errors.

As shown in the code above, I manually added the getSpanLimits() function to the Tracer prototype to make it work.

Thank you again for your time and feedback.

Reply:
Thank you for your greatly detailed explanation, but I was talking about adding a comment in the code to explain to a future developer why this function is not used but shouldn't be removed. Like so:
```suggestion
  // not used in our codebase but needed for compatibility. See issue #1244 
  getSpanLimits() {
```

Reply:
@simon-id 
Thank you for your suggestion, I will commit your suggestion.